### PR TITLE
enable use of `Context`  in tests and misc test refactoring

### PR DESCRIPTION
**Pull request by g-r-a-n-t**
- updated the test runner to enable context-related operations
- moved a handful of tests from the legacy-tests crate to the tests crate

### CODE REVIEW

File: crates/analyzer/tests/analysis.rs [hunk 1]
```diff
@@ -184,7 +184,6 @@ test_analysis! { erc20_token, "demos/erc20_token.fe"}
 test_analysis! { guest_book, "demos/guest_book.fe"}
 test_analysis! { simple_open_auction, "demos/simple_open_auction.fe"}
 test_analysis! { uniswap, "demos/uniswap.fe"}
-test_analysis! { address_bytes10_map, "features/address_bytes10_map.fe"}
```
Comment (g-r-a-n-t) [inline crates/analyzer/tests/analysis.rs:187]: Im ok with removing most of these

File: crates/test-runner/src/lib.rs [hunk 2]
```diff
@@ -68,25 +70,224 @@ impl Display for TestSink {
     }
 }
 
+// pub fn execute(name: &str, bytecode: &str, sink: &mut TestSink) -> bool {
```
Comment (g-r-a-n-t) [inline crates/test-runner/src/lib.rs:73]: delete

File: crates/test-runner/src/lib.rs [hunk 3]
```diff
@@ -68,25 +70,224 @@ impl Display for TestSink {
...
+    if result.is_success() {
         sink.inc_success_count();
     } else {
         sink.insert_failure(name, &format!("{result:?}"));
     };
 
-    reverted
+    result.is_success()
 }
+
```
Comment (g-r-a-n-t) [inline crates/test-runner/src/lib.rs:114]: delete

File: crates/tests-legacy/src/features.rs [hunk 4]
```diff
@@ -234,36 +234,36 @@ fn test_assert() {
...
-            Some(&uint_token(0)),
-        );
-
-        validate_revert(
-            harness.capture_call(&mut executor, "get_from_storage", &[uint_token(10)]),
-            &encoded_panic_out_of_bounds(),
-        );
-    })
-}
+// #[test]
```
Comment (g-r-a-n-t) [inline crates/tests-legacy/src/features.rs:237]: readd reverts portion of test

File: crates/tests/fixtures/files/call_fn.fe [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        addr: address(foo), 
+        value: 0, 
+        input_offset: MEM_PTR, 
+        input_len: 32, 
+        output_offset: MEM_PTR + 32, 
+        output_len: 32
+    ) == 1
+    assert evm::mload(offset: MEM_PTR + 32) == 0x02
+
+    // evm::mstore(offset: MEM_PTR, value: 0x03)
```
Comment (g-r-a-n-t) [inline crates/tests/fixtures/files/call_fn.fe:36]: add this part

File: crates/tests/fixtures/files/arrays.fe [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    // verify that the call reverts
+    assert evm::call(
+        gas: 1000, 
+        addr: address(array_index_oob), 
+        value: 0, 
+        input_offset: 0, 
+        input_len: 0, 
+        output_offset: 0, 
+        output_len: 0
+    ) == 0
```
Comment (cburgdorf) [inline crates/tests/fixtures/files/arrays.fe:52]: It looks like the old tests also covered that the correct error code is returned whereas these tests simply check that it reverts but do not check the actual error that was returned. Maybe we can add this back?

**Review by cburgdorf — APPROVED** [review]
Looks good, just left a comment regarding loss of assertions of revert error.

**Pull request merged**